jmxtrans-output/jmxtrans-output-kafka/src/main/java/com/googlecode/jmxtrans/model/output/kafka/KafkaWriter.java

```diff
@@ -92,10 +98,10 @@ public KafkaWriter(
 						DEFAULT_ROOT_PREFIX);
 		// Setting all the required Kafka Properties
 		Properties kafkaProperties =  new Properties();
-		kafkaProperties.setProperty("metadata.broker.list", Settings.getStringSetting(settings, "metadata.broker.list", null));
-		kafkaProperties.setProperty("zk.connect", Settings.getStringSetting(settings, "zk.connect", null));
-		kafkaProperties.setProperty("serializer.class", Settings.getStringSetting(settings, "serializer.class", null));
-		this.producer= new Producer<>(new ProducerConfig(kafkaProperties));
+		kafkaProperties.setProperty(BOOTSTRAP_SERVERS_CONFIG, Settings.getStringSetting(settings, BOOTSTRAP_SERVERS_CONFIG, null));
```

Comment:
Not sure I understand the question. I can tell it is required, because without a kafka server configured, the messages won't go anywhere. To make the code explicit about this requirement, a validation check should be added, which would raise an exception if the value isn't set.